zerver/lib/message.py

```diff
@@ -1347,10 +1358,19 @@ def apply_unread_message_event(
             user_ids_string=user_ids_string,
         )
 
+    # TODO/compatibility: Translation code for the rename of
+    # `wildcard_mentioned` to `stream_wildcard_mentioned`. Remove this
+    # when one can no longer directly upgrade from 7.x to main.
+    # This is needed for the correct `stream_wildcard_mentioned in flags`
+    # check below and updating state["mentions"] accordingly.
+    if "wildcard_mentioned" in flags:
+        flags.remove("wildcard_mentioned")
+        flags.append("stream_wildcard_mentioned")
```

Comment:
I'd be surprised if this was correct for a client that doesn't have the new client capability -- how did you verify this logic?